Documentation/ApiOverview/Seo/PageTitleApi.rst

```diff
@@ -64,11 +68,50 @@ this method you can create your own logic to define the correct title.
 
 Usage example, for example, in an :ref:`Extbase <extbase>` controller:
 
-.. code-block:: php
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/Controller/SomeController.php
+
+    $titleProvider = GeneralUtility::makeInstance(MyOwnPageTitleProvider::class);
+    $titleProvider->setTitle('Title from controller action');
+
+
+Configure the new page title provider to be used in your TypoScript setup:
+
+..  literalinclude:: _ExampleSetInController/_setup.typoscript
+    :caption: EXT:my_sitepackage/Configuration/TypoScript/setup.typoscript
+
+Example: Use website title from the site configuration
+------------------------------------------------------
+
+If you want to use data from the :ref:`site configuration <sitehandling>`, for
+example the site title, you can implement a page title provider as follows:
+
+..  literalinclude:: _ExampleWebsiteTitle/_WebsiteTitleProvider.php
+    :caption: EXT:my_sitepackage/Classes/PageTitle/WebsiteTitleProvider.php
+
+As we need to :ref:`inject <DependencyInjection>` the class :php:`SiteFinder`
+to retrieve the current site configuration, we must make the new page title
+provider public:
+
+..  literalinclude:: _ExampleWebsiteTitle/_Services.yaml
+    :caption: EXT:my_sitepackage/Configuration/Services.yaml
+
+Then **flush the cache** in :guilabel:`Admin Tools > Maintenance > Flush TYPO3
+and PHP Cache`.
+
+Configure the new page title provider to be used in your TypoScript setup:
+
+..  literalinclude:: _ExampleWebsiteTitle/_setup.typoscript
+    :caption: EXT:my_sitepackage/Configuration/TypoScript/setup.typoscript
```

Comment:
```suggestion
    :language: typoscript
    :caption: EXT:my_sitepackage/Configuration/TypoScript/setup.typoscript
```

Also: When using my_sitepackage as extension key, the namespace of the PHP classes should also be adjusted to MyVendor\MySitepackage.